### ISSUE TITLE
subMenu support?

### ISSUE THREAD

**Issue by httpete** [sparksuite/react-accessible-dropdown-menu-hook#286]
Hi, love the idea of hooks for this - but I need subMenus. Is it possible / example? Thanks

**Comment by corymharper** [comment]
Hi @httpete , thank you for the interest in our package. The hook does not currently support submenus. There are a few accessibility based keyboard controls that our hook does not take into account, specifically right / left arrow controls which should open a submenu if one such menu exists. 

As for if it such an implementation is possible using our hook, it could probably be accomplished by calling the hook multiple times, one for the menu and each sub menu, and assigning the returned props to the correct menu items, giving the `buttonProps` to the parent menuitem that should open a submenu, and so on. But, you would need to take care of the left / right arrow controls yourself, make sure to overwrite the parent menuitem that has a submenu's role from `menubutton` to `menuitem` and verify that it still met the [authoring practices](https://www.w3.org/TR/wai-aria-practices-1.2/#menu). 

In short, it is feasible but I don't recommend it and the package does not support it. I think it could be a great addition in the future for the package to support submenus, but it would take considerable work.

**Comment by httpete** [comment]
OK for anyone looking here, I found https://reakit.io/ which did this quite easily. thanks!
